auth/service.cc

```diff
@@ -403,14 +404,16 @@ future<> create_role(
         const authentication_options& options) {
     return ser.underlying_role_manager().create(name, config).then([&ser, name, &options] {
         if (!auth::any_authentication_options(options)) {
-            return make_ready_future<>();
+            return ser.underlying_role_manager().update_custom_options(name, options.role_options);
```

Comment:
As I mentioned before, I don't believe we can hijack `CREATE ROLE ... WITH OPTIONS`. It is reserved for the authenticator. See [dse](https://docs.datastax.com/en/dse/6.7/cql/cql/cql_reference/cql_commands/cqlCreateRole.html#cqlCreateRole__cqlCreateRoleoptions-optionmap-axjdHcEh).

We'll need a new keyword, `SESSION OPTIONS = map`. A more SQL compatible/traditional format is to disaggregate the map: `WITH SESSION READ TIMEOUT = duration AND SESSION WRITE TIMEOUT = duration, but it's some amount of busy work with the parser. @kostja